compiler/rustc_lint/src/builtin.rs

```diff
@@ -1295,15 +1295,80 @@ impl UnreachablePub {
         vis_span: Span,
         exportable: bool,
     ) {
+        fn opt_parent_module(tcx: TyCtxt<'_>, mut did: LocalDefId) -> Option<LocalDefId> {
```

Comment:
There's already a method for this - `tcx.parent_module_from_def_id(...)`.

Reply:
Technically they don't do the same thing. This one doesn't return an def id if the provided one is the def id of the crate root instead of returning the same def id as provided.

Not sure if that's an issue for us here, hence why I created this function here to be explicit about this.

Reply:
It's called from a place where encountering the crate root is impossible, and you have unreachable code due to the `opt_` part.

Reply:
Changed to use `tcx.parent_module_from_def_id`.